storage: TestSplitSnapshotRace_SnapshotWins failed

[(storage).TestSplitSnapshotRace_SnapshotWins failed](https://viewLog.html?buildId=1685610&tab=buildLog) on [master@07ebd0230cea54110f249420fd215adc1d2d5ecf](https://github.com/cockroachdb/cockroach/commits/07ebd0230cea54110f249420fd215adc1d2d5ecf):

```
I200113 07:05:04.146046 271753 storage/replica_destroy.go:164  [s3,r39/3:{m-/Table/Syste…}] removed 10 (1+9) keys in 4ms [clear=3ms commit=1ms]
I200113 07:05:04.370931 270290 storage/replica_command.go:1706  [s5,r39/4:{m-/Table/Syste…}] change replicas (add [] remove [(n2,s2):2VOTER_DEMOTING]): existing descriptor r39:{m-/Table/SystemConfigSpan/Start} [(n4,s4):1, (n2,s2):2, (n6,s6):5, (n5,s5):4, next=6, gen=18]
I200113 07:05:04.460892 270290 storage/replica_raft.go:248  [s5,r39/4:{m-/Table/Syste…},txn=33f7fb93] proposing ENTER_JOINT(r2 l2) REMOVE_REPLICA[(n2,s2):2VOTER_DEMOTING]: after=[(n4,s4):1 (n2,s2):2VOTER_DEMOTING (n6,s6):5 (n5,s5):4] next=6
W200113 07:05:04.507640 270290 storage/replica_write.go:201  [s5,r39/4:{m-/Table/Syste…},txn=33f7fb93] intents not processed as async resolution is disabled
I200113 07:05:04.513243 270290 storage/replica_command.go:1706  [s5,r39/4:{m-/Table/Syste…}] change replicas (add [] remove []): existing descriptor r39:{m-/Table/SystemConfigSpan/Start} [(n4,s4):1, (n2,s2):2VOTER_DEMOTING, (n6,s6):5, (n5,s5):4, next=6, gen=19]
I200113 07:05:04.564601 270290 storage/replica_raft.go:248  [s5,r39/4:{m-/Table/Syste…},txn=e29ac79c] proposing LEAVE_JOINT: after=[(n4,s4):1 (n2,s2):2LEARNER (n6,s6):5 (n5,s5):4] next=6
W200113 07:05:04.616493 270290 storage/replica_write.go:201  [s5,r39/4:{m-/Table/Syste…},txn=e29ac79c] intents not processed as async resolution is disabled
I200113 07:05:04.625473 270290 storage/replica_command.go:1706  [s5,r39/4:{m-/Table/Syste…}] change replicas (add [] remove [(n2,s2):2LEARNER]): existing descriptor r39:{m-/Table/SystemConfigSpan/Start} [(n4,s4):1, (n2,s2):2LEARNER, (n6,s6):5, (n5,s5):4, next=6, gen=20]
I200113 07:05:04.695767 270290 storage/replica_raft.go:248  [s5,r39/4:{m-/Table/Syste…},txn=ec19f9cf] proposing SIMPLE(r2) REMOVE_REPLICA[(n2,s2):2LEARNER]: after=[(n4,s4):1 (n5,s5):4 (n6,s6):5] next=6
W200113 07:05:04.743648 270290 storage/replica_write.go:201  [s5,r39/4:{m-/Table/Syste…},txn=ec19f9cf] intents not processed as async resolution is disabled
I200113 07:05:04.766134 270618 storage/store_remove_replica.go:129  [s2,r39/2:{m-/Table/Syste…}] removing replica r39/2
I200113 07:05:04.849660 270290 util/stop/stopper.go:539  quiescing
I200113 07:05:04.874021 270290 util/stop/stopper.go:539  quiescing
I200113 07:05:04.897856 270290 util/stop/stopper.go:539  quiescing
I200113 07:05:04.921465 270290 util/stop/stopper.go:539  quiescing
I200113 07:05:05.769637 274831 storage/store_snapshot.go:977  [raftsnapshot,s5,r39/4:{m-/Table/Syste…}] sending RAFT snapshot c94c053f at applied index 45
I200113 07:05:05.775362 274831 storage/store_snapshot.go:1020  [raftsnapshot,s5,r39/4:{m-/Table/Syste…}] streamed snapshot to (n4,s4):1: kv pairs: 27, log entries: 0, rate-limit: 8.0 MiB/sec, 0.06s
I200113 07:05:05.794222 274937 storage/replica_raftstorage.go:785  [s4,r39/1:{m-/Table/Syste…}] applying RAFT snapshot [id=c94c053f index=45]
I200113 07:05:05.833855 274937 storage/replica_raftstorage.go:806  [s4,r39/1:{m-/Table/Syste…}] applied RAFT snapshot [total=39ms ingestion=4@35ms id=c94c053f index=45]
E200113 07:05:05.884651 274849 storage/queue.go:1035  [gc,s4,r29/4:{/System/Max-m}] batch timestamp 1.800000125,2595 must be after replica GC threshold 1578809105.760522429,0
I200113 07:05:05.924585 275021 util/stop/stopper.go:539  quiescing
I200113 07:05:05.925668 275016 util/stop/stopper.go:539  quiescing
I200113 07:05:05.926443 275017 util/stop/stopper.go:539  quiescing
I200113 07:05:05.927107 275019 util/stop/stopper.go:539  quiescing
I200113 07:05:05.927785 275018 util/stop/stopper.go:539  quiescing
I200113 07:05:05.929145 275020 util/stop/stopper.go:539  quiescing
I200113 07:05:06.197575 275015 util/stop/stopper.go:539  quiescing
W200113 07:05:06.202305 272928 storage/raft_transport.go:637  while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.209499 273108 storage/raft_transport.go:637  while processing outgoing Raft queue to node 6: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.212120 271874 storage/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W200113 07:05:06.221315 272170 storage/raft_transport.go:637  while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.222044 273160 storage/raft_transport.go:637  while processing outgoing Raft queue to node 6: EOF:
W200113 07:05:06.223688 271328 gossip/gossip.go:1517  [n6] no incoming or outgoing connections
W200113 07:05:06.225164 272900 storage/raft_transport.go:637  while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.225808 271248 gossip/gossip.go:1517  [n5] no incoming or outgoing connections
W200113 07:05:06.226161 270556 gossip/gossip.go:1517  [n2] no incoming or outgoing connections
W200113 07:05:06.226709 271667 storage/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200113 07:05:06.224126 272178 storage/raft_transport.go:637  while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.228693 271040 gossip/gossip.go:1517  [n4] no incoming or outgoing connections
W200113 07:05:06.229542 271750 storage/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.230820 271949 storage/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200113 07:05:06.235933 271675 storage/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200113 07:05:06.236987 275015 util/stop/stopper.go:539  quiescing
I200113 07:05:06.238539 275015 util/stop/stopper.go:539  quiescing
I200113 07:05:06.240038 275015 util/stop/stopper.go:539  quiescing
I200113 07:05:06.241398 275015 util/stop/stopper.go:539  quiescing
I200113 07:05:06.243353 275015 util/stop/stopper.go:539  quiescing
I200113 07:05:06.244518 275015 util/stop/stopper.go:539  quiescing
--- FAIL: TestSplitSnapshotRace_SnapshotWins (8.71s)
    client_split_test.go:1588: batch timestamp 1.800000125,2625 must be after replica GC threshold 1578809105.760522429,0
```

<details><summary>Repro</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestSplitSnapshotRace_SnapshotWins PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(storage).TestSplitSnapshotRace_SnapshotWins failed](https://viewLog.html?buildId=1695683&tab=buildLog) on [master@96ce803dbcaecb60119589ddbb842ef5b7ddf904](https://github.com/cockroachdb/cockroach/commits/96ce803dbcaecb60119589ddbb842ef5b7ddf904):

```
I200117 00:35:50.734000 272376 storage/replica_destroy.go:164  [s3,r39/3:{m-/Table/Syste…}] removed 10 (1+9) keys in 7ms [clear=2ms commit=5ms]
W200117 00:35:50.865844 274045 storage/replica_read.go:120  [range-lookup="m",s1,r1/1:/{Min-System/NodeL…}] intents not processed as async resolution is disabled
I200117 00:35:50.970783 270850 storage/replica_command.go:1706  [s5,r39/4:{m-/Table/Syste…}] change replicas (add [] remove [(n2,s2):2VOTER_DEMOTING]): existing descriptor r39:{m-/Table/SystemConfigSpan/Start} [(n4,s4):1, (n2,s2):2, (n6,s6):5, (n5,s5):4, next=6, gen=18]
I200117 00:35:51.196052 270850 storage/replica_raft.go:248  [s5,r39/4:{m-/Table/Syste…},txn=f2918cd2] proposing ENTER_JOINT(r2 l2) REMOVE_REPLICA[(n2,s2):2VOTER_DEMOTING]: after=[(n4,s4):1 (n2,s2):2VOTER_DEMOTING (n6,s6):5 (n5,s5):4] next=6
W200117 00:35:51.251137 270850 storage/replica_write.go:202  [s5,r39/4:{m-/Table/Syste…},txn=f2918cd2] intents not processed as async resolution is disabled
I200117 00:35:51.265768 270850 storage/replica_command.go:1706  [s5,r39/4:{m-/Table/Syste…}] change replicas (add [] remove []): existing descriptor r39:{m-/Table/SystemConfigSpan/Start} [(n4,s4):1, (n2,s2):2VOTER_DEMOTING, (n6,s6):5, (n5,s5):4, next=6, gen=19]
I200117 00:35:51.445328 270850 storage/replica_raft.go:248  [s5,r39/4:{m-/Table/Syste…},txn=1e3c1e34] proposing LEAVE_JOINT: after=[(n4,s4):1 (n2,s2):2LEARNER (n6,s6):5 (n5,s5):4] next=6
W200117 00:35:51.537094 270850 storage/replica_write.go:202  [s5,r39/4:{m-/Table/Syste…},txn=1e3c1e34] intents not processed as async resolution is disabled
I200117 00:35:51.548696 270850 storage/replica_command.go:1706  [s5,r39/4:{m-/Table/Syste…}] change replicas (add [] remove [(n2,s2):2LEARNER]): existing descriptor r39:{m-/Table/SystemConfigSpan/Start} [(n4,s4):1, (n2,s2):2LEARNER, (n6,s6):5, (n5,s5):4, next=6, gen=20]
I200117 00:35:51.648483 270850 storage/replica_raft.go:248  [s5,r39/4:{m-/Table/Syste…},txn=fe9a1ee2] proposing SIMPLE(r2) REMOVE_REPLICA[(n2,s2):2LEARNER]: after=[(n4,s4):1 (n5,s5):4 (n6,s6):5] next=6
W200117 00:35:51.691051 270850 storage/replica_write.go:202  [s5,r39/4:{m-/Table/Syste…},txn=fe9a1ee2] intents not processed as async resolution is disabled
I200117 00:35:51.724522 271205 storage/store_remove_replica.go:129  [s2,r39/2:{m-/Table/Syste…}] removing replica r39/2
I200117 00:35:51.848983 270850 util/stop/stopper.go:539  quiescing
I200117 00:35:51.895936 270850 util/stop/stopper.go:539  quiescing
I200117 00:35:51.952431 270850 util/stop/stopper.go:539  quiescing
I200117 00:35:51.989904 270850 util/stop/stopper.go:539  quiescing
I200117 00:35:53.545468 275514 storage/store_snapshot.go:973  [raftsnapshot,s5,r39/4:{m-/Table/Syste…}] sending RAFT snapshot 52a8032a at applied index 45
I200117 00:35:53.559998 275514 storage/store_snapshot.go:1016  [raftsnapshot,s5,r39/4:{m-/Table/Syste…}] streamed snapshot to (n4,s4):1: kv pairs: 27, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200117 00:35:53.592950 275454 storage/replica_raftstorage.go:785  [s4,r39/1:{m-/Table/Syste…}] applying RAFT snapshot [id=52a8032a index=45]
I200117 00:35:53.634402 275454 storage/replica_raftstorage.go:806  [s4,r39/1:{m-/Table/Syste…}] applied RAFT snapshot [total=41ms ingestion=4@32ms id=52a8032a index=45]
E200117 00:35:53.863809 275264 storage/queue.go:1035  [gc,s4,r29/4:{/System/Max-m}] batch timestamp 1.800000125,2965 must be after replica GC threshold 1579131353.709880794,0
I200117 00:35:53.868065 275596 util/stop/stopper.go:539  quiescing
I200117 00:35:53.869379 275592 util/stop/stopper.go:539  quiescing
I200117 00:35:53.872226 275593 util/stop/stopper.go:539  quiescing
I200117 00:35:53.873234 275595 util/stop/stopper.go:539  quiescing
I200117 00:35:53.874141 275594 util/stop/stopper.go:539  quiescing
I200117 00:35:53.875075 275597 util/stop/stopper.go:539  quiescing
I200117 00:35:54.207518 275591 util/stop/stopper.go:539  quiescing
W200117 00:35:54.213049 272308 storage/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200117 00:35:54.215024 273665 storage/raft_transport.go:637  while processing outgoing Raft queue to node 6: rpc error: code = Unavailable desc = transport is closing:
W200117 00:35:54.216686 272329 storage/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.217797 272609 storage/raft_transport.go:637  while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.218375 272508 storage/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.220304 272391 storage/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.221385 271417 gossip/gossip.go:1518  [n3] no incoming or outgoing connections
W200117 00:35:54.224733 271092 gossip/gossip.go:1518  [n2] no incoming or outgoing connections
W200117 00:35:54.230152 272484 storage/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.231101 273630 storage/raft_transport.go:637  while processing outgoing Raft queue to node 6: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.233082 273364 storage/raft_transport.go:637  while processing outgoing Raft queue to node 5: EOF:
W200117 00:35:54.234454 273316 storage/raft_transport.go:637  while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.235700 272380 storage/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200117 00:35:54.242114 271727 gossip/gossip.go:1518  [n5] no incoming or outgoing connections
I200117 00:35:54.244306 275591 util/stop/stopper.go:539  quiescing
I200117 00:35:54.246339 275591 util/stop/stopper.go:539  quiescing
I200117 00:35:54.248219 275591 util/stop/stopper.go:539  quiescing
I200117 00:35:54.249492 275591 util/stop/stopper.go:539  quiescing
I200117 00:35:54.251390 275591 util/stop/stopper.go:539  quiescing
I200117 00:35:54.252761 275591 util/stop/stopper.go:539  quiescing
--- FAIL: TestSplitSnapshotRace_SnapshotWins (9.37s)
    client_split_test.go:1589: batch timestamp 1.800000125,2796 must be after replica GC threshold 1579131353.709880794,0
```

<details><summary>Repro</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestSplitSnapshotRace_SnapshotWins PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/storage/client_split_test.go
+++ b/pkg/storage/client_split_test.go
@@ -1405,6 +1405,7 @@ func runSetupSplitSnapshotRace(
 	sc.TestingKnobs.IntentResolverKnobs.DisableAsyncIntentResolution = true
 	// Avoid fighting with the merge queue while trying to reproduce this race.
 	sc.TestingKnobs.DisableMergeQueue = true
+	sc.TestingKnobs.DisableGCQueue = true
 	// Disable the split delay mechanism, or it'll spend 10s going in circles.
 	// (We can't set it to zero as otherwise the default overrides us).
 	sc.RaftDelaySplitToSuppressSnapshotTicks = -1

